src/key.rs

```diff
@@ -1161,12 +1207,25 @@ impl XOnlyPublicKey {
     ///
     /// Returns [`Error::InvalidPublicKey`] if the length of the data slice is not 32 bytes or the
     /// slice does not represent a valid Secp256k1 point x coordinate.
+    #[deprecated(since = "TBD", note = "Use `from_byte_array` instead.")]
     #[inline]
     pub fn from_slice(data: &[u8]) -> Result<XOnlyPublicKey, Error> {
-        if data.is_empty() || data.len() != constants::SCHNORR_PUBLIC_KEY_SIZE {
-            return Err(Error::InvalidPublicKey);
+        match <[u8; constants::SCHNORR_PUBLIC_KEY_SIZE]>::try_from(data) {
+            Ok(data) => Self::from_byte_array(&data),
+            Err(_) => Err(InvalidPublicKey),
         }
+    }
```

Comment:
This function could be written with combinators like this:

```rust
    pub fn from_slice(data: &[u8]) -> Result<XOnlyPublicKey, Error> {
        <[u8; constants::SCHNORR_PUBLIC_KEY_SIZE]>::try_from(data)
            .map_err(|_| InvalidPublicKey)
            .and_then(|data| Self::from_byte_array(&data))
    }
```

Reply:
It does look quite neat with a combinator